### PR TITLE
fix: Fixed an issue where the correct table was not focused when sharing URLs in TableDetail

### PR DESCRIPTION
**Pull request by MH4GF**
https://github.com/user-attachments/assets/e9174908-d52d-4284-a7f2-3b103d81e24f

ref: https://liam-erd-sample-5rp1jfv9b-route-06-core.vercel.app/?active=account_statuses_cleanup_policies

### CODE REVIEW

File: frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/useAutoLayout/useAutoLayout.ts [hunk 1]
```diff
@@ -1,37 +1,41 @@
...
   } = useERDContentContext()
 
-  const handleLayout = useCallback(async () => {
-    setLoading(true)
-    const nodes = getNodes()
-    const edges = getEdges()
-    const visibleNodes: Node[] = nodes.filter((node) => !node.hidden)
-    const hiddenNodes: Node[] = nodes.filter((node) => node.hidden)
+  const handleLayout = useCallback(
+    async (fitViewOptions?: FitViewOptions) => {
```
Comment (MH4GF) [inline frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/useAutoLayout/useAutoLayout.ts:14]: 📝 It is only possible to pass fitViewOptions.

File: frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/useInitialAutoLayout.ts [hunk 2]
```diff
@@ -1,14 +1,38 @@
...
+import { useERDContentContext } from './ERDContentContext'
 import { useAutoLayout } from './useAutoLayout'
 
+const getActiveTableNameFromUrl = (): string | undefined => {
+  const urlParams = new URLSearchParams(window.location.search)
+  const activeQueryParam: QueryParam = 'active'
+  const tableName = urlParams.get(activeQueryParam)
+
+  return tableName || undefined
+}
```
Comment (MH4GF) [inline frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/useInitialAutoLayout.ts:14]: 📝 The contents of `useActiveTableNameFromUrl()` have been brought in.

File: frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/useInitialAutoLayout.ts [hunk 3]
```diff
@@ -1,14 +1,38 @@
...
 
   useEffect(() => {
+    if (initializeComplete) {
+      return
+    }
+
+    const tableNameFromUrl = getActiveTableNameFromUrl()
+    updateActiveTableName(tableNameFromUrl)
+    const fitViewOptions = tableNameFromUrl
+      ? { maxZoom: 1, duration: 300, nodes: [{ id: tableNameFromUrl }] }
```
Comment (hoshinotsuyoshi) [inline frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/useInitialAutoLayout.ts:31]: 📝  same as useFitViewWhenActiveTableChange

**Review by hoshinotsuyoshi — APPROVED** [review]
LGTM!

**Pull request merged**